File: src/toolchain/workload/platforms/android/z42vm/src/main/java/io/z42/vm/Z42VMModule.kt
package io.z42.vm

/**
 * Opaque handle for a loaded `.zbc` module. Lifetime tied to the
 * owning [Z42VM]; once the VM is closed, holding a [Z42VMModule]
 * is allowed but any operation through it surfaces as
 * [Z42VMException.NOT_INIT].
 */
class Z42VMModule internal constructor(
    @JvmField internal val nativeHandle: Long,
)

File: src/toolchain/workload/platforms/ios/Sources/Z42VM/Z42VMModule.swift
import Foundation

/// Opaque handle for a loaded `.zbc` module. Lifetime is tied to its
/// owning `Z42VM`; once the VM is `deinit`'d, holding a `Z42VMModule`
/// is allowed but any operation through it returns `.notInit`.
public final class Z42VMModule {
    let handle: OpaquePointer

    init(handle: OpaquePointer) {
        self.handle = handle
    }
}
